pkg/scaler/scaling_test.go

```diff
@@ -635,5 +749,86 @@ var _ = Describe("Node scaling tests", func() {
 			Expect(err).NotTo(HaveOccurred())
 			Expect(result).To(BeTrue())
 		})
+
+	})
+
+	Context("Handle Drain Strategy", func() {
+
+		It("Execute function return error", func() {
+			var node1Name = "test-node-1"
+			nodes := &corev1.NodeList{
+				Items: []corev1.Node{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: node1Name,
+						},
+					},
+					{},
+				},
+			}
+			mockDrainStrategy := mockDrain.NewMockNodeDrainStrategy(mockCtrl)
+			strategy := []*drain.DrainStrategyResult{{Message: "TEST", HasExecuted: true}, {}}
+
+			gomock.InOrder(
+				mockDrainStrategy.EXPECT().Execute(gomock.Any(), gomock.Any()).Return(strategy, fmt.Errorf("error in execute function")),
+				// mockDrainStrategy.EXPECT().HasFailed(gomock.Any(), gomock.Any()).Return(true, fmt.Errorf("error in Hasfailed function")),
+			)
+			result, err := handleDrainStrategy(mockKubeClient, mockDrainStrategy, *nodes, logger)
+			Expect(err).Should(HaveOccurred())
+			Expect(result).To(BeFalse())
+		})
+		It("Handle Drain strategy has failed", func() {
+
+			var node1Name = "test-node-1"
+
+			nodes := &corev1.NodeList{
+				Items: []corev1.Node{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: node1Name,
+						},
+					},
+					{},
+				},
+			}
+			mockDrainStrategy := mockDrain.NewMockNodeDrainStrategy(mockCtrl)
+			strategy := []*drain.DrainStrategyResult{{Message: "TEST", HasExecuted: true}, {}}
+
+			gomock.InOrder(
+				mockDrainStrategy.EXPECT().Execute(gomock.Any(), gomock.Any()).Return(strategy, nil).AnyTimes(),
+				mockDrainStrategy.EXPECT().HasFailed(gomock.Any(), gomock.Any()).Return(true, nil),
+			)
+			result, err := handleDrainStrategy(mockKubeClient, mockDrainStrategy, *nodes, logger)
+			Expect(err).Error()
+			Expect(result).To(BeFalse())
+		})
+		It("Handle Drain strategy has failed", func() {
```

Comment:
Unit case name is duplicate here, we can add more specific test name ex. Handle Drain strategy has failed due to timeout

Reply:
Updated the testcase name.